### ISSUE TITLE
code_type enumeration gives false positive

### ISSUE THREAD

**Issue by jburse** [SWI-Prolog/swipl-devel#474]
Hi,

Was trying on Windows platform:

```
Welcome to SWI-Prolog (threaded, 64 bits, version 8.1.6)
SWI-Prolog comes with ABSOLUTELY NO WARRANTY. This is free software.

?- code_type(X, white).
X = 9 ;
X = 32 ;
X = 65545 
```
But the codepoint 65545 is from [Mycenaean Greek](https://en.wikipedia.org/wiki/Linear_B_Syllabary).

**Comment by jburse** [comment]
I checked the source code of pl-ctype.c:

```
static int
iswhite(wint_t chr)
{ return chr == ' ' || chr == '\t';
}
```
So probably a higher order bit truncation problem.

**Comment by JanWielemaker** [comment]
Thanks.  Fixed with e1f2e893116356b4aa25aade058d6001b52f9ab0
